docs/ForDevelopers/Dependencies.md

```diff
@@ -24,45 +29,115 @@ For other recent Linux distributions, you can also rely on the pre-built binarie
-RUN  cmake --build build --target install -j$(nproc)
+cmake --build build --target install -j$(nproc)
+```
+
+</TabItem>
+<TabItem value="macos" label="MacOS">
+Install LLVM/MLIR using Homebrew:
+
+```shell
+brew install llvm@20
 ```
 
+### Binaries
+⚠️ **Caution**: the pre-built binaries provided by the LLVM project on the Github release pages **DO NOT** serve as a replacement, since they lack the required MLIR support.
+
+### Building from Source
+
+1. Install XCode (through the App Store).
+2. Install the build requisites: `brew install cmake ninja z3`
+2. Make sure to replace all occurrences of [install_prefix] with your preferred install prefix of LLVM.
+```shell
+wget https://github.com/llvm/llvm-project/releases/download/llvmorg-20.1.4/llvm-project-20.1.4.src.tar.xz
+tar -xf llvm-project-20.1.4.src.tar.xz
+mkdir llvm-project-20.1.4.src/build
+cd llvm-project-20.1.4.src
+env VIRTUAL_ENV=/venv cmake -B build  -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;mlir;polly;lldb" -DLLVM_ENABLE_RUNTIMES="compiler-rt;libcxx;libcxxabi;libunwind;pstl;openmp" -DLLVM_TARGETS_TO_BUILD="AArch64" -DLLVM_BUILD_EXAMPLES=OFF -DCMAKE_BUILD_TYPE=Debug -G Ninja -DLLVM_ENABLE_ASSERTIONS=ON  -DLLVM_BUILD_TESTS=OFF -DLLVM_BUILD_LLVM_DYLIB=ON -DLLVM_LINK_LLVM_DYLIB=ON -DLLVM_ENABLE_DUMP=ON -DLLVM_ENABLE_FFI=ON -DCMAKE_CXX_FLAGS="-fno-omit-frame-pointer -mno-omit-leaf-frame-pointer" -DLLVM_PARALLEL_LINK_JOBS=1 -DLLVM_PARALLEL_TABLEGEN_JOBS=10 -DBUILD_SHARED_LIBS=OFF -DLLVM_INSTALL_UTILS=ON  -DLLVM_ENABLE_ZLIB=OFF -DLLVM_POLLY_LINK_INTO_TOOLS=ON -DLLVM_BUILD_EXTERNAL_COMPILER_RT=ON -DLLVM_ENABLE_EH=OFF -DLLVM_ENABLE_RTTI=ON -DLLVM_INCLUDE_DOCS=OFF -DLLVM_INCLUDE_TESTS=OFF -DLLVM_OPTIMIZED_TABLEGEN=ON -DLLVM_USE_RELATIVE_PATHS_IN_FILES=ON -DLLVM_SOURCE_PREFIX=. -DLLDB_USE_SYSTEM_DEBUGSERVER=ON -DLIBOMP_INSTALL_ALIASES=OFF -DLIBCXX_INSTALL_MODULES=ON -DLLVM_CREATE_XCODE_TOOLCHAIN=OFF -DCLANG_FORCE_MATCHING_LIBCLANG_SOVERSION=OFF -DLLVM_BUILD_LLVM_C_DYLIB=ON -DLLVM_ENABLE_LIBCXX=ON -DLIBCXX_PSTL_BACKEND=libdispatch -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_PROJECT_TOP_LEVEL_INCLUDES=/opt/homebrew/Library/Homebrew/cmake/trap_fetchcontent_provider.cmake -Wno-dev -DBUILD_TESTING=OFF -DCMAKE_OSX_SYSROOT=$(xcrun --sdk macosx --show-sdk-path) -DLLVM_ENABLE_Z3_SOLVER=ON -DFFI_INCLUDE_DIR=$(xcrun --sdk macosx --show-sdk-path)/usr/include/ffi -DFFI_LIBRARY_DIR=$(xcrun --sdk macosx --show-sdk-path)/usr/lib -DLIBCXX_INSTALL_LIBRARY_DIR=[install_prefix]/lib/c++ -DLIBUNWIND_INSTALL_LIBRARY_DIR=[install_prefix]/lib/unwind -DLIBCXXABI_INSTALL_LIBRARY_DIR=[install_prefix]/c++ -DRUNTIMES_CMAKE_ARGS="-DCMAKE_INSTALL_RPATH=@loader_path|@loader_path/../unwind" -DBUILTINS_CMAKE_ARGS="-DCOMPILER_RT_ENABLE_IOS=OFF;-DCOMPILER_RT_ENABLE_WATCHOS=OFF;-DCOMPILER_RT_ENABLE_TVOS=OFF" -DCMAKE_PREFIX_PATH=/opt/homebrew -DCMAKE_INSTALL_PREFIX=[install_prefix] llvm/
```

Comment:
In my environment, compiling nearly every file in LingoDB takes more than one second. This might be due to the build configuration flag currently in use.

For context, I did not install llvm@20 via Homebrew. Instead, I am using clang and clang++ that were built with the following configuration:

ini
복사
편집
-DCMAKE_BUILD_TYPE=Debug
Would switching to -DCMAKE_BUILD_TYPE=Release help improve the build performance?